library/src/scripts/components/forms/select/SelectOne.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import Select from "react-select";
+import { getRequiredID, IOptionalComponentID } from "@library/componentIDs";
+import classNames from "classnames";
+import { t } from "@library/application";
+import menuList from "@library/components/forms/select/overwrites/menuList";
+import menu from "@library/components/forms/select/overwrites/menu";
+import selectContainer from "@library/components/forms/select/overwrites/selectContainer";
+import doNotRender from "@library/components/forms/select/overwrites/doNotRender";
+import Paragraph from "@library/components/Paragraph";
+import { IComboBoxOption } from "./BigSearch";
+import SelectOption from "@library/components/forms/select/overwrites/selectOption";
+import { IFieldError } from "@library/@types/api";
+import ErrorMessages from "@dashboard/components/forms/ErrorMessages";
+import valueContainer from "@library/components/forms/select/overwrites/valueContainer";
+import controlContainer from "@library/components/forms/select/overwrites/controlContainer";
+
+interface IProps extends IOptionalComponentID {
+    label: string;
+    disabled?: boolean;
+    className?: string;
+    placeholder?: string;
+    options: IComboBoxOption[];
+    setData: (data: any) => void;
+    labelNote?: string;
+    noteAfterInput?: string;
+    errors?: IFieldError[];
+    searchable?: boolean;
+}
+
+/**
+ * Implements the search bar component
+ */
+export default class SelectOne extends React.Component<IProps> {
+    private id: string;
+    private prefix = "SelectOne";
+    private inputID: string;
+    private errorID: string;
+
+    constructor(props: IProps) {
+        super(props);
+        this.id = getRequiredID(props, this.prefix);
+        this.inputID = this.id + "-input";
+        this.errorID = this.id + "-errors";
+    }
+
+    private handleOnChange = (newValue: any, actionMeta: any) => {
+        this.props.setData(newValue);
+    };
+
+    public render() {
+        const { className, disabled, options, searchable } = this.props;
+
+        /** The children to be rendered inside the indicator. */
+        const componentOverwrites = {
+            IndicatorsContainer: doNotRender,
+            SelectContainer: selectContainer,
```

Comment:
Let's pull this out into a class variable. It doesn't seem to change between renders.

If it did then we could make it a getter, but generally I think it's best to try and keep the render methods as small and straightforward as possible.